AppCenterPush/AppCenterPush/MSPush.m

```diff
@@ -389,4 +392,10 @@ - (BOOL)didReceiveRemoteNotification:(NSDictionary *)userInfo fromUserNotificati
   return NO;
 }
 
+- (void)authTokenContext:(__unused MSAuthTokenContext *)authTokenContext didReceiveAuthToken:(__unused NSString *)authToken {
+  if (self.pushToken) {
```

Comment:
Unless you can guarantee it's always same thread accessing pushToken (and that would deserve comments), accessing self.pushToken after a if is not thread safe.
Either mark pushToken atomic and use a local variable before the if or use synchronized.